fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java

```diff
@@ -1269,29 +1277,46 @@ public void executeAndSendResult(boolean isOutfileQuery, boolean isSendFields,
                     // will be recognized as a success result
                     // so We need to send fields after first batch arrived
                     if (!isSendFields) {
-                        if (!isOutfileQuery) {
-                            sendFields(queryStmt.getColLabels(), queryStmt.getFieldInfos(),
-                                    getReturnTypes(queryStmt));
-                        } else {
+                        if (isOutfileQuery) {
                             if (!Strings.isNullOrEmpty(queryStmt.getOutFileClause().getSuccessFileName())) {
                                 outfileWriteSuccess(queryStmt.getOutFileClause());
                             }
                             sendFields(OutFileClause.RESULT_COL_NAMES, OutFileClause.RESULT_COL_TYPES);
+                        } else if (isBlackHoleClause) {
+                            // do nothing
+                        } else {
+                            sendFields(queryStmt.getColLabels(), queryStmt.getFieldInfos(),
+                                    getReturnTypes(queryStmt));
                         }
                         isSendFields = true;
                     }
-                    for (ByteBuffer row : batch.getBatch().getRows()) {
-                        channel.sendOnePacket(row);
+                    if (isBlackHoleClause) {
+                        // For blackhole queries, aggregate data by BE nodes
+                        Map<String, String> attachedInfos = batch.getBatch().getAttachedInfos();
+                        if (attachedInfos != null && !attachedInfos.isEmpty()) {
```

Comment:
there are too many if else for BlackHoleSink, maybe we could impl like this:
1. let BlackHoleSink as a abstract class, which could return the customized header
2. let WarmupSink impl the BlackHoleSink
3. always use one sink instance for WarmupSink, it merge all be's result and add a line for total
4. remove all if-else for BalckHoleSink because it could work as normal query